src/main/java/controllers/FeedPresenter.java

```diff
@@ -34,4 +51,54 @@ public void setFeedFilter(String filter) {
     public void refreshFeed() {
         this.feedManager.refreshFeed();
     }
+
+    public String getFilterInput() {
+        // Will implement restrictions later to the input that the user is allowed to type.
+        try {
+            String filterInput = this.inOut.getInput(
+                    "What type of cuisine would you like to see in your feed?" +
+                            "Type 'None' if you do not wish to have a filter.");
+            return filterInput;
+        } catch(IOException e) {
+            return "That cuisine type does not exist here.";
+        }
+    }
+
+    public void setOutputToInOut(ArrayList<ArrayList<String>> feedInStrings) {
+        for (ArrayList<String> postInStrings : feedInStrings) {
+            for (String postDetails : postInStrings) {
+                this.inOut.setOutput(postDetails);
+            }
+            this.inOut.setOutput(System.getProperty("line.separator"));
+        }
+    }
+
+    public ArrayList<ArrayList<String>> convertFeedToString(Feed feed) {
+        ArrayList<ArrayList<String>> feedList = new ArrayList<ArrayList<String>>();
```

Comment:
Since we know the length of the lists here, lets use a nested array: String[][].